ExtraCredit/ExtraCredit/ExtraCredit.vcxproj

```
@@ -0,0 +1,233 @@
+    </ClCompile>
+    <Link>
+      <SubSystem>Console</SubSystem>
+      <GenerateDebugInformation>true</GenerateDebugInformation>
+      <AdditionalDependencies>%(AdditionalDependencies);DerivedShapes.dll</AdditionalDependencies>
+      <AdditionalLibraryDirectories>$(SolutionDir)DLL_A;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+    </Link>
+    <PreBuildEvent>
+      <Command>
+      </Command>
+    </PreBuildEvent>
+    <PostBuildEvent>
+      <Command>
+      </Command>
+    </PostBuildEvent>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
+    <ClCompile>
+      <WarningLevel>Level3</WarningLevel>
+      <FunctionLevelLinking>true</FunctionLevelLinking>
+      <IntrinsicFunctions>true</IntrinsicFunctions>
+      <SDLCheck>true</SDLCheck>
+      <PreprocessorDefinitions>WIN32;NDEBUG;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <ConformanceMode>true</ConformanceMode>
+      <AdditionalIncludeDirectories>$(SolutionDir)Shapes\src;$(SolutionDir)UnitTest\src;$(SolutionDir)DerivedShapes;$(SolutionDir)Shapes\x64\Debug;$(SolutionDir)DerivedShapes\x64\Debug</AdditionalIncludeDirectories>
+    </ClCompile>
+    <Link>
+      <SubSystem>Console</SubSystem>
+      <EnableCOMDATFolding>true</EnableCOMDATFolding>
+      <OptimizeReferences>true</OptimizeReferences>
+      <GenerateDebugInformation>true</GenerateDebugInformation>
+      <AdditionalDependencies>DerivedShapes.lib;DerivedShapes.dll;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalLibraryDirectories>..\..\DerivedShapes\x64\Debug;..\..\Shapes\x64\Debug;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+    </Link>
+    <PreBuildEvent>
+      <Command>
+      </Command>
+    </PreBuildEvent>
+    <PostBuildEvent>
+      <Command>
+      </Command>
+    </PostBuildEvent>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+    <ClCompile>
+      <WarningLevel>Level4</WarningLevel>
+      <SDLCheck>true</SDLCheck>
+      <PreprocessorDefinitions>_DEBUG;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <ConformanceMode>true</ConformanceMode>
+      <AdditionalIncludeDirectories>$(SolutionDir)Shapes\src;$(SolutionDir)UnitTest\src;$(SolutionDir)DIIA\DerivedShapes;$(SolutionDir)DIIB\DerivedShapes;$(SolutionDir)DLL_A;$(SolutionDir)DLL_B</AdditionalIncludeDirectories>
+    </ClCompile>
+    <Link>
+      <SubSystem>Console</SubSystem>
+      <GenerateDebugInformation>true</GenerateDebugInformation>
+      <AdditionalDependencies>%(AdditionalDependencies);DerivedShapes.dll</AdditionalDependencies>
+      <AdditionalLibraryDirectories>$(SolutionDir)DLL_A;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```

```suggestion
+      <AdditionalDependencies>DerivedShapes.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalLibraryDirectories>$(SolutionDir)DLL_A\$(Platform)\$(Configuration);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```
In AdditionalDependencies, specify by .lib rather than .dll, even though it is a DLL; when DLLs are built, a .lib (different from a static-library .lib) is also created, and is the thing to link with (it ends up pointing to the DLL).

AdditionalLibraryDirectories changes to align with the changes I proposed in your DerivedShapes.vcxproj's.

---

and of course, these would apply to all 4 build configurations, not just Debug|x64